### ISSUE TITLE
Docker Desktop Instability with WSL or Hyper-V

### ISSUE THREAD

**Issue by Platf0rmz** [docker/for-win#8306]
- [X] I have tried with the latest version of my channel (Stable or Edge)  <-- I have tried with Stable and Edge
  - [X] I have uploaded Diagnostics
  - Diagnostics ID:  B7E90471-5133-4962-A23D-DE75E5D620D3/20200901162224

### Expected behavior

Docker Desktop WSL 2 backend should stay up and stable.

### Actual behavior

Windows notification pop-up at the bottom right of the screen shows this message.

"Docker Desktop WSL 2 backend has stopped unexpectedly.  Do you want to restart it?  -Docker Desktop"

### Information

  - Windows Version:   I have tried this with a Windows 10 1909 and 2004 build.
  - Docker Desktop Version:  I have tried with latest edge, v19.03.13-beta2 and with stable v19.03.12
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM:  Running in Windows.  I have tried with using the WSL-2 backend or the older Hyper-V backend.

If running with the Hyper-V backend, Docker will also eventually fail requiring me to restart Docker Desktop.

In both the Hyper-V and WSL backends, the message while trying to show the server versions at the time of failure from a "docker version" command is "Error response from daemon:  open \\.\pipe\docker_engine_linux:  The system cannot find the file specified."

It however shows the Client: Docker Engine - Community statistics.  Again, it will just not show the "Server:  Docker Engine - Community" statistics instead replacing it with the error response from daemon message I just described.

### Steps to reproduce the behavior

1.  Start docker.
2.  Just wait about 10-15 minutes before it will fail.  There is no need to start a container as it will fail on its own.

**Comment by simonferquel** [comment]
If you stop docker desktop and run a wsl terminal, does it crash as well after 20-25 mins ?

**Comment by Platf0rmz** [comment]
I was thinking the same thing to try so I ended up doing that but taking it a step further.  I uninstalled Docker Desktop to eliminate that variable.  I then opened up an Ubuntu terminal and ran a simple infinite loop to list my /proc/meminfo stats every 3 seconds.  In parallel outside of Ubuntu via Powershell, I ran another infinite loop to give me the output of 'wsl -l -v' every 3 seconds so I quickly can see if the distro stopped or is running.  In as little as 3 minutes, my Ubuntu session quit and the state of the distro went from running to stopped.  I'm going to try to parse logs and events to see WTF could be going on...........
